scr/main/java/bittercode/model/StoreException.java

```diff
@@ -0,0 +1,65 @@
+package com.bittercode.model;
+
+import java.io.IOException;
+
+import com.bittercode.constant.ResponseCode;
+
+public class StoreException extends IOException {
+
+    private String errorCode;
+    private String errorMessage;
+    private int statusCode;
+
+    public StoreException(String errorMessage) {
+        super(errorMessage);
+        this.errorCode = "BAD_REQUEST";
+        this.setStatusCode(400);
+        this.errorMessage = errorMessage;
+    }
+
+    public StoreException(ResponseCode errorCodes) {
+        super(errorCodes.getMessage());
+        this.statusCode = errorCodes.getCode();
+        this.errorMessage = errorCodes.getMessage();
+        this.setErrorCode(errorCodes.name());
+    }
+
+    public StoreException(String errroCode, String errorMessage) {
+        super(errorMessage);
+        this.errorCode = errroCode;
+        this.errorMessage = errorMessage;
+        this.statusCode = 422;
+    }
+
+    public StoreException(int statusCode, String errorCode, String errorMessage) {
+        super(errorMessage);
+        this.statusCode = statusCode;
+        this.errorCode = errorCode;
+        this.errorMessage = errorMessage;
+    }
+
+    public String getErrorCode() {
+        return null;
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Fix getErrorCode method return value</b></div>
<div id="fix">

The `getErrorCode()` method always returns `null` instead of returning the `errorCode` field value. This could lead to unexpected behavior when error codes are needed.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -41,3 +41,3 @@
  public String getErrorCode() {
 -        return null;
 +        return errorCode;
      }
```


```suggestion
public String getErrorCode() {
        return errorCode;
    }
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/7#issuecomment-2527741727>#f9242c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged